encoding/encoding.go

```diff
@@ -51,6 +52,20 @@ type Serializer interface {
 	CheckSuccess(body *transport.Response) error
 }
 
+// StreamRequestReader interface exposes method to read multiple request body
+type StreamRequestReader interface {
+	// NextBody returns the encoded request body if available, and if not, returns an
+	// io.EOF error
```

Comment:
The comment here is a bit weird. Interface definition should not mention it is a `io.EOF` error since this sounds like an implementation details.
It should just say, `if not, returns error why it is unavailable`